TESTS/mbed_drivers/timeout/timeout_tests.h

```diff
@@ -201,8 +201,8 @@ void test_no_wait(void)
         Semaphore sem(0, 1);
         T timeout;
         timeout.attach_callback(mbed::callback(sem_callback, &sem), 0ULL);
-        int32_t sem_slots = sem.wait(0);
-        TEST_ASSERT_EQUAL(1, sem_slots);
+        bool sem_sts = sem.try_acquire_for(0);
```

Comment:
```suggestion
        bool sem_acquired = sem.try_acquire_for(0);
        TEST_ASSERT_EQUAL(true, sem_acquired);
```
I suggest using a more meaningful variable name.

Reply:
Idiomatically, it's better to use `try_acquire()` rather than `try_acquire_for(0)`.

Doesn't matter much if mixing it with timed stuff, but in principle, `try_acquire()` gives you scope to omit timing code from bare metal builds, whereas the `_for` form forcibly includes it.